### PR TITLE
Update to clang-format 12

### PR DESCRIPTION
**Pull request by paulb777**
clang-format 11 has disappeared from brew causing the style check in the check CI workflow to fail.

This PR upgrades to clang-format 12.0.0 which does some weird things with lines including `NS_SWIFT_NAME` including disregarding the 100 character line length.

clang-format bugs reported at https://bugs.llvm.org/show_bug.cgi?id=49995 and https://bugs.llvm.org/show_bug.cgi?id=49996

Any suggestions here would be appreciated.

### CODE REVIEW

File: Firebase/InstanceID/Public/FIRInstanceID.h
Comment (ryanwilson) [inline Firebase/InstanceID/Public/FIRInstanceID.h:27]: Not overly worried about these since they'll go away shortly.

File: FirebaseDatabase/Sources/Core/FWriteTree.m
Comment (ryanwilson) [inline FirebaseDatabase/Sources/Core/FWriteTree.m:259]: 😳  this condition is.... something. Probably fine to not touch it for safety reasons but yowza.

File: FirebasePerformance/Sources/Public/FIRHTTPMetric.h
Comment (ryanwilson) [inline FirebasePerformance/Sources/Public/FIRHTTPMetric.h:39]: I think this is egregious enough to warrant a `#pragma` to ignore formatting.

**Review by ryanwilson — COMMENTED** [review]
I think I'm personally okay with leaving most of them long, otherwise wrapping them with the same comments you added in Firestore makes sense to me. Not ideal though by any means.

File: FirebasePerformance/Sources/Public/FIRHTTPMetric.h
Comment (ncooke3) [reply FirebasePerformance/Sources/Public/FIRHTTPMetric.h:39]: it made a stair*case* of enum *case*s  🤣

File: FirebaseABTesting/Sources/Public/FirebaseABTesting/FIRLifecycleEvents.h
Comment (ryanwilson) [inline FirebaseABTesting/Sources/Public/FirebaseABTesting/FIRLifecycleEvents.h:23]: Nit: since these are back to back, you could mention it doesn't honour line length for `NS_SWIFT_NAME` and leave it off all the way until line 34

File: FirebaseAppDistribution/Sources/Public/FirebaseAppDistribution/FIRAppDistribution.h
Comment (ryanwilson) [inline FirebaseAppDistribution/Sources/Public/FirebaseAppDistribution/FIRAppDistribution.h:70]: Same mention here of just leaving it off until line 77

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuthErrors.h
Comment (ryanwilson) [inline FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuthErrors.h:72]: Same comment here

**Review by ryanwilson — APPROVED** [review]
LGTM with a small nit - your call if you want to change it or leave it as-is.

**Review by ryanwilson — APPROVED** [review]

**Comment by paulb777** [comment]
I created https://bugs.llvm.org/show_bug.cgi?id=49995 and https://bugs.llvm.org/show_bug.cgi?id=49996 to the llvm project

**Comment by mydeveloperday** [comment]
See comments on https://reviews.llvm.org/D89425

**Comment by paulb777** [comment]
@mydeveloperday Thanks for the pointer. cc: @bhamiltoncx

**Pull request merged**
